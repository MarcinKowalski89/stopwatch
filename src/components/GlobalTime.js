import React from 'react';

const GlobalTime = () => (
  <div>
    <h1>Global Time!</h1>
  </div>
);

export default GlobalTime;
