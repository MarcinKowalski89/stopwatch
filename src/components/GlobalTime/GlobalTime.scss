.row {
  .list-clocks {
    width: 200px;
    overflow: auto;
    max-height: 94%;
    position: fixed;
    left: 0;
    top: 45px;

    li {
      list-style: none;
      padding: 5px;
      cursor: pointer;
    }
  }


  .clocks-container {
    position: absolute;
    right: 0;
    width: calc(100% - 200px);
    .clock-wrapper {

      .clock {
        position: relative;
        border-radius: 50%;
        height: 250px;
        width: 250px;
        margin: 15px;
        background-color: #eee;
        border: 2px solid #000;
        box-shadow: 0 0 5px #000;

        &::after,
        .hours,
        .minutes,
        .seconds {
          position: absolute;
        }

        &::after,
        .hours,
        .minutes {
          background-color: #000;
        }

        .hours,
        .minutes {
          transform-origin: 50% 100%;
        }

        &::after {
          content: '';
          background: red;
          border-radius: 50%;
          left: 50%;
          top: 50%;
          transform: translate(-50%,-50%);
          width: 15px;
          height: 15px;
        }

        .hours {
          height: 20%;
          left: 48.75%;
          top: 30%;
          width: 2.5%;
        }

        .minutes {
          height: 40%;
          left: 49%;
          top: 10%;
          width: 2%;
        }

        .seconds {
          background: red;
          height: 45%;
          left: 49.5%;
          top: 14%;
          transform-origin: 50% 80%;
          width: 1%;
        }
        .btn-danger {
          position: absolute;
          right: -10px;
          top: -10px;
        }
      }

      .city {
        text-align: center;
      }
    }
  }

}
