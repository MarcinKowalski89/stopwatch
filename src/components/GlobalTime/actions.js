export const storeClocks = citys => ({
  type: 'STORE_CLOCKS',
  data: citys,
});
export const showClock = city => ({
  type: 'SHOW_CLOCK',
  data: city,
});
export const hideClock = city => ({
  type: 'HIDE_CLOCK',
  data: city,
});
