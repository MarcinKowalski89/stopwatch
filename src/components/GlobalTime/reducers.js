const initialState = {};
export default (state = initialState, action) => {
  switch (action.type) {
    case 'STORE_CLOCKS':
      action.data.forEach((element) => { initialState[element.zoneName.split('/').splice(-1)] = element; });
      return initialState;
    case 'SHOW_CLOCK':
      return { ...state, [action.data]: { ...state[action.data], show: true } };
    case 'HIDE_CLOCK':
      return { ...state, [action.data]: { ...state[action.data], show: false } };
    default:
      return state;
  }
};
