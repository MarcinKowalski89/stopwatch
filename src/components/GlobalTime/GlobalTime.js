import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import { showClock } from './actions';
import Clock from '../Clock/Clock';
import './GlobalTime.scss';

const GlobalTime = props => (
  <Fragment>
    <ul className="d-flex flex-row">
      {Object.keys(props.clocks).map(clock => (
        <li key={clock}>
          <button
            className="btn btn-info"
            onClick={() => props.showClock(clock)}
            disabled={props.clocks[clock].show}
          >
            {props.clocks[clock].city}
          </button>
        </li>
      ))}
    </ul>
    <div className="d-flex flex-wrap">
      {Object.keys(props.clocks).map(clock => (
        props.clocks[clock].show
          ? <Clock
            key={clock}
            city={clock}
          />
          : null
      ))}
    </div>
  </Fragment>
);

GlobalTime.propTypes = { clocks: PropTypes.objectOf(Object).isRequired };

export default connect(
  state => ({
    clocks: state.globalTime,
  }),
  {
    showClock,
  },
)(GlobalTime);
