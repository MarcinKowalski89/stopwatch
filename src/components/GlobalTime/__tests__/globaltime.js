import React from 'react';
import { shallow } from 'enzyme';

import { GlobalTime } from '../GlobalTime';

test('GlobalTime should match default snapshot', () => {
  const component = shallow(<GlobalTime
    clocks={{
      Warsaw: {
        city: 'Warsaw',
        locale: 'en-US',
        timeZone: 'Europe/Warsaw',
        show: false,
      },
      NewYork: {
        city: 'New York',
        locale: 'en-US',
        timeZone: 'America/New_York',
        show: false,
      },
      Tallinn: {
        city: 'Tallinn',
        locale: 'en-US',
        timeZone: 'Europe/Tallinn',
        show: false,
      },
      Dubai: {
        city: 'Dubai',
        locale: 'en-US',
        timeZone: 'Asia/Dubai',
        show: false,
      },
    }}
    hideClock={() => null}
  />);

  expect(component).toMatchSnapshot();
});

test('GlobalTime should match snapshot with Warsaw and Dubai clocks', () => {
  const component = shallow(<GlobalTime
    clocks={{
      Warsaw: {
        city: 'Warsaw',
        locale: 'en-US',
        timeZone: 'Europe/Warsaw',
        show: true,
      },
      NewYork: {
        city: 'New York',
        locale: 'en-US',
        timeZone: 'America/New_York',
        show: false,
      },
      Tallinn: {
        city: 'Tallinn',
        locale: 'en-US',
        timeZone: 'Europe/Tallinn',
        show: false,
      },
      Dubai: {
        city: 'Dubai',
        locale: 'en-US',
        timeZone: 'Asia/Dubai',
        show: true,
      },
    }}
    hideClock={() => null}
  />);

  expect(component).toMatchSnapshot();
});
