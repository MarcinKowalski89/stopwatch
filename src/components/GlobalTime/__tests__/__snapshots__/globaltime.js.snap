// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GlobalTime should match default snapshot 1`] = `
<React.Fragment>
  <ul
    className="d-flex flex-row"
  >
    <li
      key="Warsaw"
    >
      <button
        className="btn btn-info"
        disabled={false}
        onClick={[Function]}
      >
        Warsaw
      </button>
    </li>
    <li
      key="NewYork"
    >
      <button
        className="btn btn-info"
        disabled={false}
        onClick={[Function]}
      >
        New York
      </button>
    </li>
    <li
      key="Tallinn"
    >
      <button
        className="btn btn-info"
        disabled={false}
        onClick={[Function]}
      >
        Tallinn
      </button>
    </li>
    <li
      key="Dubai"
    >
      <button
        className="btn btn-info"
        disabled={false}
        onClick={[Function]}
      >
        Dubai
      </button>
    </li>
  </ul>
  <div
    className="d-flex flex-wrap"
  />
</React.Fragment>
`;

exports[`GlobalTime should match snapshot with Warsaw and Dubai clocks 1`] = `
<React.Fragment>
  <ul
    className="d-flex flex-row"
  >
    <li
      key="Warsaw"
    >
      <button
        className="btn btn-info"
        disabled={true}
        onClick={[Function]}
      >
        Warsaw
      </button>
    </li>
    <li
      key="NewYork"
    >
      <button
        className="btn btn-info"
        disabled={false}
        onClick={[Function]}
      >
        New York
      </button>
    </li>
    <li
      key="Tallinn"
    >
      <button
        className="btn btn-info"
        disabled={false}
        onClick={[Function]}
      >
        Tallinn
      </button>
    </li>
    <li
      key="Dubai"
    >
      <button
        className="btn btn-info"
        disabled={true}
        onClick={[Function]}
      >
        Dubai
      </button>
    </li>
  </ul>
  <div
    className="d-flex flex-wrap"
  >
    <Clock
      city="Warsaw"
      hideClock={[Function]}
      key="Warsaw"
      locale="en-US"
      show={true}
      timeZone="Europe/Warsaw"
    />
    <Clock
      city="Dubai"
      hideClock={[Function]}
      key="Dubai"
      locale="en-US"
      show={true}
      timeZone="Asia/Dubai"
    />
  </div>
</React.Fragment>
`;
